src/panels/calendar/recurrence.ts

```diff
@@ -138,3 +166,127 @@ export function ruleByWeekDay(
     }
   });
 }
+
+/**
+ * Determine the recurrence options based on the day of the month. The
+ * return values are a Weekday object that represent a BYDAY for a
+ * particular week of the month like "first Saturday" or "last Friday".
+ */
+function getWeekydaysForMonth(dtstart: Date): Weekday[] {
+  const weekDay = getWeekday(dtstart);
+  const dayOfMonth = getDate(dtstart);
+  const nthWeekdayOfMonth = Math.floor((dayOfMonth - 1) / 7) + 1;
+  const isLastWeekday = !isSameMonth(dtstart, addDays(dtstart, 7));
+  const byweekdays: Weekday[] = [];
+  if (!isLastWeekday || dayOfMonth <= 28) {
+    byweekdays.push(new Weekday(weekDay, nthWeekdayOfMonth));
+  }
+  if (isLastWeekday) {
+    byweekdays.push(new Weekday(weekDay, -1));
+  }
+  return byweekdays;
+}
+
+/**
+ * Returns the list of repeat values available for the specified date.
+ */
+export function getMonthlyRepeatItems(
+  hass: HomeAssistant,
+  interval: number,
+  dtstart: Date
+): MonthlyRepeatItem[] {
+  const getLabel = (repeatValue: string) =>
+    renderRRuleAsText(hass, `FREQ=MONTHLY;INTERVAL=${interval};${repeatValue}`);
+
+  const result: MonthlyRepeatItem[] = [
+    // The default repeat rule is on day of month e.g. 3rd day of month
+    {
+      value: `BYMONTHDAY=${getDate(dtstart)}`,
+      label: getLabel(`BYMONTHDAY=${getDate(dtstart)}`)!,
+    },
+    // Additional optional rules based on the week of month e.g. 2nd sunday of month
+    ...getWeekydaysForMonth(dtstart).map((item) => ({
+      value: `BYDAY=${item.toString()}`,
+      byday: item,
+      label: getLabel(`BYDAY=${item.toString()}`)!,
+    })),
+  ];
+  if (isLastDayOfMonth(dtstart)) {
```

Comment:
Is this handled by `getWeekydaysForMonth` above when it does the `isLastWeekday` check?

Reply:
I don't understand?

Reply:
That is the last weekday of the month, as in last friday of the month.

This is the last day of the month, so the 28, 29, 30 or 31.